File: android/src/main/java/com/reactnativenewcanvas/NewCanvas.kt
package com.reactnativenewcanvas

class NewCanvas(private val newCanvasView: NewCanvasView) {
  var height: Int? = null
  var width: Int? = null

  private var context2d: NewCanvasContext2d? = null

  fun getContext(contextType: String): NewCanvasContext2d? {
    if (contextType.toLowerCase() === "2d") {
      if (context2d === null) {
        context2d = NewCanvasContext2d(this, newCanvasView);
      }
      return context2d;
    }
    return null;
  }
}

File: ios/NewCanvas.swift
class NewCanvas {

    var height: Int?
    var width: Int?

    private var context2d: NewCanvasContext2d?
    private var newCanvasView: NewCanvasView

    init(newCanvasView: NewCanvasView) {
        self.newCanvasView = newCanvasView
    }

    func getContext(withType contextType: String) -> NewCanvasContext2d? {
        if (contextType == "2d") {
            if (context2d == nil) {
                context2d = NewCanvasContext2d(newCanvas: self, newCanvasView: newCanvasView)
            }
            return context2d
        }
        return nil
    }
}
